### PR TITLE
Type a few more components in common-adapters

### PR DESCRIPTION
**Pull request by akalin-keybase**
- Button
- ClickableBox
- Dropdown

Move some one-off components out of common-adapters. Mark some uncommonly used components, too.

### CODE REVIEW

File: shared/common-adapters/clickable-box.desktop.js [hunk 1]
```diff
@@ -42,7 +42,10 @@ class ClickableBox extends React.Component<Props & {children: any}, {mouseDown:
     let underlay
 
     if (this.state.mouseIn) {
-      const borderRadius = style && style.borderRadius
+      let borderRadius = 0
+      if (style && typeof style === 'object') {
```
Comment (akalin-keybase) [inline shared/common-adapters/clickable-box.desktop.js]: Not sure if this is the best way to do this

Comment (chrisnojima) [reply shared/common-adapters/clickable-box.desktop.js]: we shouldn't be doing this kind of check anymore. Reaching into the style object is super dirty. We do this in only in a couple of places where we kind of 'split' the styles out and apply some to the container we make and some into the actual component. really we should have multiple style objects we expose

File: shared/styles/index.desktop.js [hunk 2]
```diff
@@ -106,14 +106,9 @@ export const backgroundURL = (...to: Array<string>) => {
...
 export const styleSheetCreate = (obj: Object) => obj
-export const collapseStyles = (styles: Array<Object> | Object) => {
-  if (isArray(styles)) {
-    return styles.reduce((map, item) => {
-      return {...map, ...item}
-    }, {})
-  } else {
-    return styles
-  }
+export const collapseStyles = (styles: $ReadOnlyArray<CollapsibleStyle>): Object => {
```
Comment (chrisnojima) [inline shared/styles/index.desktop.js]: why is this in 3 different prs?  i don't understand how all these reconcile against each other

File: shared/common-adapters/clickable-box.desktop.js (hunk 1 above)
Comment (akalin-keybase) [reply shared/common-adapters/clickable-box.desktop.js]: Okay. I'll try and fix this in a follow-up PR. `Icon` also has a similar thing with `hoverColor` that I'll try to fix.

File: shared/styles/index.desktop.js (hunk 2 above)
Comment (akalin-keybase) [reply shared/styles/index.desktop.js]: Sorry, I had rebased this PR and https://github.com/keybase/client/pull/11151 on top of the `collapseStyles` PR to make sure that it type-checked -- not sure why GitHub is still showing this as a diff, but now that I merged the `collapseStyles` PR in, it should disappear.

**Pull request merged**